modulemd/v2/modulemd-module.c

```diff
@@ -181,23 +181,68 @@ modulemd_module_init (ModulemdModule *self)
 }
 
 
-void
-modulemd_module_set_defaults (ModulemdModule *self, ModulemdDefaults *defaults)
+ModulemdDefaultsVersionEnum
+modulemd_module_set_defaults (ModulemdModule *self,
+                              ModulemdDefaults *defaults,
+                              ModulemdDefaultsVersionEnum index_mdversion,
+                              GError **error)
 {
-  g_return_if_fail (MODULEMD_IS_MODULE (self));
+  g_autoptr (ModulemdDefaults) upgraded_defaults = NULL;
+  g_autoptr (GError) nested_error = NULL;
+  g_return_val_if_fail (MODULEMD_IS_MODULE (self), MD_DEFAULTS_VERSION_ERROR);
 
   g_clear_object (&self->defaults);
-
   if (defaults == NULL)
-    return;
+    {
+      /* If we are empty here, return the passed-in index_mdversion so the
+       * function reports success and does not influence further upgrades.
+       */
+      return index_mdversion;
```

Comment:
This does not match the documentation in the header file: `Returns MD_DEFAULTS_VERSION_UNSET if @defaults was NULL`.

I think I would suggest to go with explicitly returning `VERSION_UNSET` and handling that in the index functions, so that in the future it's easier to handle this case specifically if needed.

Reply:
Thanks for catching that. One too many times back-and-forth on how to handle that case. Initially, I had been setting `MD_DEFAULT_VERSION_ERROR` to be `MD_DEFAULT_VERSION_LATEST + 1`, but then I reworked it so that it would be -1, updated the documentation and then forgot to edit that location in the code.